scripts/tests/AUTH/test_3b_key_migration_execute.py

```
@@ -0,0 +1,527 @@
+        print(f"[ERROR] Failed to create user: {response.status_code} - {response.text}")
+        return None
+    except requests.exceptions.RequestException as e:
+        print(f"[ERROR] Create user request failed: {e}")
+        return None
+
+
+def run_docker_sql(sql: str, db_path: str = "/data/sanctum.db") -> str:
+    """Run SQL inside Docker container and return output."""
+    escaped_sql = sql.replace("'", "'\\''")
+    cmd = f"docker compose exec -T backend sqlite3 -json {shlex.quote(db_path)} '{escaped_sql}'"
+    result = subprocess.run(cmd, shell=True, capture_output=True, text=True, cwd=REPO_ROOT)
```

Ruff (0.14.14)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)